source/common/http/path_utility.h

```diff
@@ -13,6 +13,8 @@ class PathUtil {
   // Returns if the normalization succeeds.
   // If it is successful, the param will be updated with the normalized path.
   static bool canonicalPath(HeaderEntry& path_header);
+  // Merges two or more adjacent slashes into one.
```

Comment:
I'd explicitly mention that this excludes query params and fragments. Although, I'm unsure if the path captures fragments anyway.